File: Coding Interviews/01 - Interview Flow/E1324.CountPrimes.py
class Solution:
    """
    @param n: a integer
    @return: return a integer
    """
    def countPrimes(self, n):
        # write your code here
        if n < 3:
            return 0

        result, hash = 0, [True] * n
        for num in range(2, n):
            if not hash[num]:
                continue

            result += 1
            for j in range(num * num, n, num):
                hash[j] = False

        return result

File: Coding Interviews/01 - Interview Flow/E1324.CountPrimes.java
public class Solution {
    /**
     * @param n: a integer
     * @return: return a integer
     */
    public int countPrimes(int n) {
        // write your code here
        if (n < 2) return 0;

        int count = 0;
        boolean[] notPrime = new boolean[n];

        for (int i = 2; i < n; i++) {
            if (notPrime[i]) continue;

            count++;
            for (int j = 2; i * j < n; j++)
                notPrime[i * j] = true;
        }

        return count;
    }
}
